sdk/eventhub/event-hubs/src/partitionPump.ts

```diff
@@ -39,7 +39,12 @@ export class PartitionPump {
   async start(): Promise<void> {
     this._isReceiving = true;
     try {
-      await this._partitionProcessor.initialize();
+      await this._partitionProcessor.initialize({
+        setStartPosition: (startPosition) => {
+          // the user can override the start position if they choose
+          this._initialEventPosition = startPosition;
```

Comment:
I think this is only meant to be used if a checkpoint doesn't exist. Right now this completely overrides it.

Reply:
Just wanted to make a note that I saw this and will fix it - I didn't realize this was only intended to be a fallback.